hook.py

```diff
@@ -1,90 +1,18 @@
-        check_call(['git', 'clone', '--depth', '1', repo_url, repo_dir], stdout=DEVNULL, stderr=STDOUT)
-
-    at_ingested = 0
-    at_total = 0
-    errors = 0
-    for filename in glob.iglob('plugins/atomic/atomic-red-team/atomics/**/T*.yaml', recursive=True):
-        for entries in BaseWorld.strip_yml(filename):
-            for test in entries['atomic_tests']:
-                try:
-                    incr_ingested, incr_total = await _save_ability(data_svc, entries, test)
-                    at_ingested += incr_ingested
-                    at_total += incr_total
-                except Exception as e:
-                    logging.debug("ERROR:", filename, e)
-                    errors += 1
-    errors_output = f" and ran into {errors} errors" if errors else ""
-    logging.debug(f'Ingested {at_ingested} abilities (out of {at_total}) from Atomic plugin{errors_output}')
-
-
-def _use_default_inputs(test, string):
-    defaults = {key: str(val["default"]) for key, val in test.get("input_arguments", {}).items()}
-    while RE_VARIABLE.search(string):
-        full_string, varname = RE_VARIABLE.search(string).groups()
-        string = string.replace(full_string, defaults[varname])
-
-    return string
-
-
-def _handle_multiline_commands(cmd):
-    return cmd.replace("\n", ";")
-
-
-async def _prepare_executor(test):
-    command = _use_default_inputs(test, test['executor']['command'])
-    command = _handle_multiline_commands(command)
-    cleanup = _use_default_inputs(test, test['executor'].get('cleanup_command', ''))
-    cleanup = _handle_multiline_commands(cleanup)
-
-    encoded_command = BaseWorld.encode_string(command)
-    encoded_cleanup = BaseWorld.encode_string(cleanup)
-
-    return (encoded_command, encoded_cleanup)
-
-
-async def _save_ability(data_svc, entries, test):
-    ability_id = hashlib.md5(json.dumps(test).encode()).hexdigest()
-    at_ingested = 0
-    at_total = 0
-    for p in test['supported_platforms']:
-        at_total += 1
-        if test['executor']['name'] == 'manual':
-            # this test is expected to be run manually by a human, no automation is provided
-            continue
-        encoded_command, encoded_cleanup = await _prepare_executor(test)
-        await data_svc.store(
-            Ability(ability_id=ability_id, tactic='redcanary', technique_id=entries['attack_technique'],
-                    technique=entries['display_name'], name=test['name'], description=test['description'],
-                    platform=PLATFORMS.get(p, 'unknown'), executor=EXECUTORS.get(test['executor']['name'], 'unknown'),
-                    test=encoded_command, cleanup=encoded_cleanup, requirements=[], parsers=[], variations=[])
-        )
-        at_ingested += 1
-
-    return (at_ingested, at_total)
+    # we only ingest data once, and save new abilities in the data/ folder of the plugin
+    if not (os.path.exists(data_dir) and os.listdir(data_dir)):
```

Comment:
we have a .gitkeep on the data/ directory. so on a fresh clone, this looks like it will not execute.. thoughts?

Reply:
fixed in 0948d05, now we look whether the `abilities` directory has been created